docs/DirectoryManager.md

```diff
@@ -0,0 +1,127 @@
+# Directory Manager
+
+This module is responsible for managing the file-system state and also provides some low level APIs for managing managed files.
+
+## VBA Directory System
+
+VBA has quite a few differences on macOS and windows systems. That is VBA is not fully supported on macOS.
+
+### `CurDir()`
+
+Inituitively the current directory `CurDir()` API should return the path of the Active Workbook. However, it appears that on Windows, the `CurDir()` needn't be the path of the active workbook. This [thread](https://stackoverflow.com/questions/41273872/excel-curdir-function-returns-a-wrong-directory) explains some details on this. The behavior of `CurDir()` is different on macOS. Moreover the state of `CurDir()` persists across macro calls and seem to work even across programs.
+
+Most VBA programs seem to not use `CurDir()` directly. They either do one of the following things -
+
+**Get absolute path using `ActiveWorkbook.Path`:**
+
+`FileName = ActiveWorkbook.Path & "\myfile.xlsx`
+
+> `ActiveWorkbook.path` is the parent directory of the current active workbook
+
+**Or set the current working directory to `ActiveWorkbook.Path`:**
+
+`ChDir (ActiveWorkbook.path)`
+
+> Note: This usage indicates that developers are not using the functionality of `CurDir()` directly. And t
+> hey are not using the persistence ability of the `CurDir()` API.
+
+> `CurDir("D")` returns the current working directory on `D Drive`. The current working drive itself can be changed by `ChDrive("D")`.
```

Comment:
Are you referring to apps script here? Mention so, it is confusing.

Reply:
I rearranged it, so it should be clear now.